File: app/Models/Course.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

use Illuminate\Database\Eloquent\Factories\HasFactory;

class Course extends Model
{
    use HasFactory;

    protected $fillable = [
        'title',
        'category',
        'price',
        'duration',
        'description',
        'image',
        'rating',           // Kriteria C2
        'students_count',   // Kriteria C3
        'difficulty_level'  // Kriteria C5
    ];

    public function students()
{
    return $this->belongsToMany(User::class, 'enrollments')
                ->withPivot('progress', 'status', 'last_accessed_at')
                ->withTimestamps();
}

    public function lessons()
    {
        return $this->hasMany(Lesson::class);
    }
}

File: skillconnect_mobile/lib/src/models/course.dart
import 'package:flutter/material.dart';

class Course {
  const Course({
    required this.title,
    required this.category,
    required this.description,
    required this.price,
    required this.rating,
    required this.students,
    required this.durationWeeks,
    required this.matchScore,
    required this.owned,
    required this.progress,
    required this.colors,
  });

  final String title;
  final String category;
  final String description;
  final int price;
  final double rating;
  final int students;
  final int durationWeeks;
  final double matchScore;
  final bool owned;
  final int progress;
  final List<Color> colors;
}
